File: loja/pedidos/src/main/java/br/com/bota/resource/PedidoResource.java
package br.com.bota.resource;

import br.com.bota.dto.PedidoDTO;
import br.com.bota.facade.PedidoFacade;
import io.micronaut.http.HttpResponse;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.PathVariable;
import io.micronaut.http.annotation.Post;
import io.micronaut.tracing.annotation.ContinueSpan;
import io.micronaut.tracing.annotation.NewSpan;
import io.micronaut.tracing.annotation.SpanTag;

import javax.inject.Inject;
import java.util.List;
import java.util.UUID;
import java.util.stream.Collectors;

@Controller("/pedido")
public class PedidoResource {
    @Inject
    private PedidoFacade facade;

    @Post
    @ContinueSpan
    public HttpResponse<PedidoDTO> criar(@SpanTag("pedido.criar") PedidoDTO dto) {
        var toSave = facade.criar(PedidoDTO.toEntity(dto));
        return HttpResponse.ok(PedidoDTO.toDTO(toSave));
    }

    @Get("/{clienteId}")
    @ContinueSpan
    public HttpResponse<List<PedidoDTO>> findByClienteId(@SpanTag("cliente.get") @PathVariable("clienteId") UUID id) {
        var resultados = facade.findBy(id);
        return HttpResponse.ok(resultados.stream().map(PedidoDTO::toDTO).collect(Collectors.toList()));
    }
}

File: loja-kotlin/kpedidos/src/main/kotlin/br/com/bota/resource/PedidoResource.kt
package br.com.bota.resource

import br.com.bota.converter.PedidoConverter
import br.com.bota.facade.PedidoFacade
import br.com.bota.lojalib.dto.PedidoDTO
import io.micronaut.http.HttpResponse
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.micronaut.http.annotation.PathVariable
import io.micronaut.http.annotation.Post
import io.micronaut.tracing.annotation.ContinueSpan
import io.micronaut.tracing.annotation.SpanTag
import jakarta.inject.Inject
import java.util.*


@Controller("/pedido")
open class PedidoResource @Inject constructor(private val facade: PedidoFacade) {


    @Post
    @ContinueSpan
    open fun criar(@SpanTag("pedido.criar") dto: PedidoDTO): HttpResponse<PedidoDTO> {
        var pedidoConvertido = facade.criar(PedidoConverter.toPedido(dto))
        return HttpResponse.created(PedidoConverter.toPedidoDTO(pedidoConvertido))
    }

    @Get("/{clienteId}")
    @ContinueSpan
    open fun findByClienteId(@SpanTag("cliente.get") @PathVariable("clienteId") id: UUID): HttpResponse<List<PedidoDTO>> {
        val resultados = facade.findById(id)
        return HttpResponse.ok(resultados.map { PedidoConverter.toPedidoDTO(it) })
    }
}